web3/contract.py

```diff
@@ -275,6 +275,28 @@ def deploy(cls, transaction=None, args=None, kwargs=None):
         txn_hash = cls.web3.eth.sendTransaction(deploy_transaction)
         return txn_hash
 
+    @classmethod
+    def constructor(cls, args=None, kwargs=None):
+        """
+        :param args: The contract constructor arguments as positional arguments
+        :param kwargs: The contract constructor arguments as keyword arguments
+        :return: a contract constructor object
+        """
+        if cls.bytecode is None:
+            raise ValueError(
+                "Cannot call constructor on a contract that does not have 'bytecode' associated "
+                "with it"
+            )
+        # is bytecode not None required by estimateGas and other methods?
```

Comment:
`bytecode` is not required for `estimateGas`, `call`, or `transact` on deployed contracts, so checking for a bytecode value here is good, since it wont happen further down.